x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/search/aggregations/GeoLineBucketedSort.java

```diff
@@ -53,7 +53,13 @@ public long sizeOf(long bucket) {
         }
         long start = inHeapMode(bucket) ? rootIndex : (rootIndex + getNextGatherOffset(rootIndex) + 1);
         long end = rootIndex + bucketSize;
-        return end - start;
+        long size = 0;
+        for (long index = start; index < end; index++) {
```

Comment:
It isn't clear to me, from a performance perspective, if I should avoid doing this loop.

Currently it is done three times.

- circuit-breaker accounting
- getPoints
- getSortValues

I can call it once and re-use the value, but I think the code would be more complicated.

Reply:
You can certainly profile it and find out. But I think you only need this in the `buildAggregation` part so its not super performance sensitive.

Reply:
thanks. that was my thought as well!